src/entities/vision/vision.cpp

```diff
@@ -0,0 +1,121 @@
+/***
+ * Maracatronics Robotics
+ * Federal University of Pernambuco (UFPE) at Recife
+ * http://www.maracatronics.com/
+ *
+ * This file is part of Armorial project.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ***/
+
+#include "vision.h"
+
+#include <include/messages_robocup_ssl_wrapper.pb.h>
+#include <include/messages_robocup_ssl_detection.pb.h>
+#include <src/utils/text/text.h>
+
+Vision::Vision(QString visionAddress, quint16 visionPort) : Entity(ENT_VISION) {
+    // Creating socket and setting network data
+    _visionClient = new QUdpSocket(this);
+    _visionAddress = visionAddress;
+    _visionPort = visionPort;
+
+    // Bind and connect socket in network
+    bindAndConnect();
+}
+
+Vision::~Vision() {
+    // Closing and deleting socket
+    if(_visionClient->isOpen()) {
+        _visionClient->close();
+    }
+
+    delete _visionClient;
+}
+
+void Vision::initialization() {
+    std::cout << Text::cyan("[VISION] ", true) << Text::bold("Started at address '" + _visionAddress.toStdString() + "' and port '" + std::to_string(_visionPort) + "'.") << '\n';
+}
+
+void Vision::loop() {
+    while(_visionClient->hasPendingDatagrams()) {
+        // Creating auxiliary vars
+        SSL_WrapperPacket wrapperData;
+        char *buffer = new char[65535];
+        long long int packetLength = 0;
+
+        // Reading upcoming packet and parsing it to protobuf
+        packetLength = _visionClient->readDatagram(buffer, 65535);
+        if(wrapperData.ParseFromArray(buffer, int(packetLength)) == false) {
+            std::cout << Text::cyan("[VISION] ", true) << Text::red("Wrapper packet parsing error.", true) << '\n';
+            continue;
+        }
+
+        // Debug vision frame
```

Comment:
Currently we're only debugging the data here, I've checked with the grSim and everything looks great